### ISSUE TITLE
Accessibility: Dark mode table header cells lack sufficient color contrast (4.05)

### ISSUE THREAD

**Issue by lancegliser** [themesberg/flowbite-react#1309]
- [X] I have searched the [Issues](https://github.com/themesberg/flowbite-react/issues) to see if this bug has already been reported
- [X] I have tested the latest version

## Steps to reproduce

1. Go to https://www.flowbite-react.com/docs/components/table
2. Turn on dark mode
3. Open dev tools
4. Inspect table header cells

## Current behavior

The current table component headers cells use `dark:text-gray-400` on top of `dark:bg-gray-700`.

## Expected behavior

Accessibility standards for table headers pass AA standards.

## Context

If the text us updated to use `dark:text-gray-300` the result is 6.99 ratio passing accessibility standards.

If possible, provide a live example URL, screenshot, video, or a repository with the minimal reproduction of the issue.
<img width="858" alt="image" src="https://github.com/themesberg/flowbite-react/assets/12085479/99214dec-51cb-467f-8456-999dcd88dc70">
